chia/full_node/mempool.py

```diff
@@ -70,10 +64,10 @@ def remove_from_pool(self, items: List[bytes32]) -> None:
             dic = self.sorted_spends[item.fee_per_cost]
             if len(dic.values()) == 0:
                 del self.sorted_spends[item.fee_per_cost]
-            self.total_mempool_cost -= item.cost
-            assert self.total_mempool_cost >= 0
-            mempool_info = self.get_mempool_info()
-            self.fee_estimator.remove_mempool_item(mempool_info, item)
+            current_mempool_cost = self.mempool_info.current_mempool_cost - item.cost
+            self.mempool_info = dataclasses.replace(self.mempool_info, current_mempool_cost=current_mempool_cost)
```

Comment:
this is a bit awkward. it looks like this should be a member function on `mempool_info`

Reply:
Yes, this part could definitely be better. I might be able to split this struct up.

I'll think about this part and update this patch.

Reply:
I moved `current_mempool_cost` / `total_mempool_cost` back to the way it was, and capture it only when we need to send the value down into the Fee Estimator. This way should work much better.